src/main/java/com/iexec/core/task/update/TaskUpdateRequestManager.java

```diff
@@ -50,17 +45,22 @@
     private static final int TASK_UPDATE_THREADS_POOL_SIZE = Runtime.getRuntime().availableProcessors() * 2;
 
     private final ExecutorService executorService = Executors.newFixedThreadPool(1);
-    private final BlockingQueue<Task> queue = createQueue();
-
     private final ConcurrentMap<String, Object> locks = ExpiringMap.builder()
             .expiration(LONGEST_TASK_TIMEOUT, TimeUnit.HOURS)
             .build();
-    private final ThreadPoolTaskExecutor taskUpdateExecutor = TaskExecutorUtils.newThreadPoolTaskExecutor(
-            "task-update-",
-            TASK_UPDATE_THREADS_POOL_SIZE);
-    private final ThreadPoolTaskExecutor requestWaitingExecutor = TaskExecutorUtils.newThreadPoolTaskExecutor(
-            "request-waiting-",
-            1);
+
+    final TaskUpdatePriorityBlockingQueue queue = new TaskUpdatePriorityBlockingQueue();
+    // Both `corePoolSize` and `maximumPoolSize` should be set to `TASK_UPDATE_THREADS_POOL_SIZE`.
+    // Otherwise, `taskUpdateExecutor` won't pop `maximumPoolSize` threads
+    // as new threads are popped only if the queue is full
+    // - which never happens with an unbounded queue.
+    final ThreadPoolExecutor taskUpdateExecutor = new ThreadPoolExecutor(
+            TASK_UPDATE_THREADS_POOL_SIZE,
+            TASK_UPDATE_THREADS_POOL_SIZE,
+            60,
```

Comment:
* @param keepAliveTime when the number of threads is greater than
     *        the core, this is the maximum time that excess idle threads
     *        will wait for new tasks before terminating.
    
 Why do we need that?

Reply:
That seems to be the default behavior - with default value of 60 seconds. However, it is unused in our implementation: we never exceed the core pool size as the maximum pool size is equal to the core and the queue is unbounded.

As far as I understand the `ThreadPoolExecutor`, it works as follows:
- always have `corePoolSize` active threads, even if they do nothing;
- if there is too much work for the `corePoolSize` threads - i.e. the queue is full -, new threads are popped until `maximumPoolSize` threads are reached;
- if there has been too much work at a point and there's now not that much work, some threads are killed to free memory/processors after a period - default to 60 seconds.
It's an automatically scalability process, useful for lots of case but not there :)

Reply:
Yes, from my understanding we could set an explicit 0 (https://github.com/JetBrains/jdk8u_jdk/blob/94318f9185757cc33d2b8d527d36be26ac6b7582/src/share/classes/java/util/concurrent/Executors.java#L88).

Reply:
You're right. Fixed in https://github.com/iExecBlockchainComputing/iexec-core/pull/479/commits/a25e077b0cd01d8a0da68dce2b5f96a6c046328f.